cmd/govim/buffer_events.go

```diff
@@ -17,6 +18,19 @@ import (
 
 func (v *vimstate) bufReadPost(args ...json.RawMessage) error {
 	nb := v.currentBufferInfo(args[0])
+	rel, err := filepath.Rel(v.workingDirectory, nb.URI().Filename())
+	if err != nil {
+		v.Logf("bufReadPost: failed to call filepath.Rel(%q, %q): %v", v.workingDirectory, nb.URI().Filename(), err)
+	}
+
+	// If we load a buffer that already had diagnostics reported by gopls, the buffer number must be
+	// updated to ensure that sign placement etc. works.
+	for i, d := range v.diagnosticsCache {
+		if d.Buf == -1 && d.Filename == rel {
```

Comment:
I think the identification of the buffer should be on the full path, not the relative path. That way we make this part of the code at least agnostics of the cwd.

By all means calculate and cache the current rel path during lazy evaluation as a separate field

Reply:
Indeed, I think we even talked about it but it must have slipped my mind, apologies.